src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs

```diff
@@ -3253,6 +3249,8 @@ internal void FinishContinuations()
             lock (continuations) { }
             int continuationCount = continuations.Count;
 
+            bool etwIsEnabled = log.IsEnabled();
```

Comment:
Are we concerned that `IsEnabled()` may have changed in-between the above check and now? I assume that is why the existing code is written the way it was.

cc @stephentoub

Reply:
I was actually looking at the existing cases and we are pretty inconsistent on this

Reply:
Why not store this Boolean at the top of the method, where it was storing log in a variable?

Reply:
> I assume that is why the existing code is written the way it was.

Yes.  The goal was to avoid output the subsequent events if and only if the starting event was written.